components/brave_new_tab_ui/containers/newTab/settings/search.tsx

```diff
@@ -79,8 +80,8 @@ export default function SearchSettings() {
 
         // If we've just enabled the searchbox, make sure at least one engine
         // is enabled.
-        if (e.checked) {
-          maybeEnableDefaultEngine()
+        if (e.checked && !hasEnabledEngine()) {
+          setEngineEnabled({ host: 'search.brave.com' }, true)
```

Comment:
use `braveSearchHost` from `config.ts` (you'll need to export it)
```suggestion
          setEngineEnabled({ host: braveSearchHost }, true)
```